internal/context/context.go

```diff
@@ -0,0 +1,41 @@
+package context
+
+import (
+	"context"
+)
+
+type ctxKey string
+
+const componentCtxKey = ctxKey("component")
+
+type RequestOrginatorType int
+
+const (
+	Dispatcher RequestOrginatorType = iota
+	EventWorkflow
+	SessionWorkflow
+	ScheduleWorkflow
+	Middleware
+	Unknown
+)
+
+func (c RequestOrginatorType) String() string {
+	if c >= Unknown {
+		return "unknown"
+	}
+	return [...]string{"dispatcher", "eventsWF", "sessionWF", "schedulerWF", "middleware"}[c]
```

Comment:
use the full name
WF is not a good candidate
sessionsWorkflow
schedulerWorkflow

middleware -> api_call

(middleware has no real meaning)